operator/src/network/controller.rs

```diff
@@ -30,22 +29,35 @@ use tracing::{debug, error, trace};
 use crate::network::{
     bootstrap, cas,
     ceramic::{self, CeramicConfig},
-    peers, utils, BootstrapSpec, CeramicSpec, Network, NetworkStatus,
+    peers,
+    utils::{HttpRpcClient, RpcClient},
+    BootstrapSpec, CeramicSpec, Network, NetworkStatus,
 };
 
 //can add resource apis here
-struct ContextData {
-    client: Client,
+pub struct Context<R> {
```

Comment:
what is having both HttpRpcClient and  RpcClient for?

Reply:
The `rpc_client` is a client to the IPFS RPC endpoint. The `RpcClient` is the trait and `HttpRpcClient` implements the trait (using http). Having the `RpcClient` trait means we can mock it out in tests and not have to spin up IPFS directly. However `IpfsRpcClient` is likely a better name.

Reply:
ok that makes sense, and the mocks make sense